### ISSUE TITLE
Cannot connect - Mac OS

### ISSUE THREAD

**Issue by alexw668** [deltaflyer4747/Xiaomi_Yi#8]
Hi,

I have a Yi camera. I cloned this Github project and tried the python script CC.pyw on my MacBook Pro, and when the UI comes up, it shows IP address, JSON port, etc. But after clicking the Connect C&C button, a window pops up that says "Cannot connect to the address specified". 

Is the problem of my Yi camera or the Python script itself?

I also tried to enable telnet and shell console access to the camera (following instructions on https://forum.dashcamtalk.com/threads/xiaomi-console-access.11198/) without any success. I wonder if the problem is related. 

Basically, I want to find a way to set its timezone to be something that's different from the Beijing Time.

Thanks for any suggestion!
Alex

**Comment by deltaflyer4747** [comment]
You have to connect to camera's wifi first or use camera client mode.

**Comment by alexw668** [comment]
Thanks for the quick response. My camera is currently connected to my wifi (so I can run the iOS app from Xiaomi). Is it the same wifi you are talking about? How do I use or get into camera client mode?

Thanks again,
Alex

**Comment by deltaflyer4747** [comment]
If you are connected to the camera's wifi (= ie you can run official app), just make sure your IP & ports are 192.168.42.1 7878 8787 80

https://forum.dashcamtalk.com/threads/camera-wifi-in-client-mode-working-ash-script-inside.12239/

**Comment by alexw668** [comment]
I tried that IP address and all three ports, but I can't connect to the camera. I simply want to just change the timezone of the camera. Do you know a way of doing it? I was hoping your python program would provide a way to connect to the camera and have a way of doing it.

Appreciate your help.

Alex

**Comment by deltaflyer4747** [comment]
if you are connected with your computer to your camera, the app should work. However i am not aware, that the computer can run iOS apps for iphone... So your previous statement - that you are able to connect with your iOS app - is misleading. The app looks for the camera's wifi network and connects to it, camera itself does NOT connect to your network!

**Comment by alexw668** [comment]
Sorry for the confusion. 

I can run the iOS app because my camera can be connected to my WiFi. I tried to run the python app in two ways: with my camera directly connected to my MacBookPro (via USB cable), or leave the camera on (connected to my wifi network). Neither worked. I also tried telnet in both ways. 

As I said, I simply want to change the timezone of the OS running on the camera. I wish Xiaomi's mobile app (or any app from them) can do that, but they don't have that.

Thanks,
Alex

**Comment by deltaflyer4747** [comment]
it does not matter what do you want to do with the camera, you have to be connected VIA WIFI on your computer first. This camera does not support USB controlling. And as i said - iOS app RECONNECTS to DIFFERENT wifi network than your home network.

**Comment by deltaflyer4747** [comment]
And if you didn't (obviously) use the client mode, the camera is NOT connected to your home wifi network, yet creates its own hotspot.

**Comment by alexw668** [comment]
I think I get it. I will try figuring it. 

Thanks for your time.

Alex

**Comment by deltaflyer4747** [comment]
Just look for the wifi network with name starting with "YDXJ" and connect to it, password is 1234567890

**Comment by alexw668** [comment]
Thanks for reopening the issue. I assume I need to reset the camera before I can see YDXJ, correct? Right now, when it's powered on, it connects to my wifi automatically (with blue light on), and I don't see a wifi network with name YDXJ when my machine search for wifi network.

Alex

**Comment by deltaflyer4747** [comment]
if you didn't change the wifi name via app, then no. But factory reset 
won't hurt.

Dne 17.8.2015 v 22:48 Alex Wang napsal(a):

**Comment by javl** [comment]
@alexw668 To make it clear: the camera does NOT connect to your home WIFI. When you turn on the camera, it creates its own WIFI network. As soon as the blue light on the side starts blinking, this network is available for your phone to connect to. When you start the app, your phone will disconnect from your home network and connect to the network provided by the camera.

So what @deltaflyer4747 is trying to say is you need to make sure your computer is connected to the WIFI network created by the camera, not your home network.
By default, this network name starts with `YDXJ`, but maybe you have changed this in the settings, thinking you had to provide your home network details? If that is the case, it means there are now two networks with the same name, which would be confusing. Resetting to factory default settings might be the easiest way to set the network name back to its original value (with the `YDXJ` part). After that, open your computer's network settings and connect to the camera's network with the `1234567890` password.

**Comment by deltaflyer4747** [comment]
Actually there is a script that allows you to connect your cammera to your home network (does not work with latest firmware, i have to fix it - and for that i have to find some free time...)

**Comment by javl** [comment]
I know, but I didn't mention it as @alexw668 didn't seem to be using that code.
